types/amazon-connect-streams/index.d.ts

```diff
@@ -0,0 +1,688 @@
+// Type definitions for non-npm package Amazon Connect Streams API 1.3
+// Project: https://github.com/aws/amazon-connect-streams
+// Definitions by: Andy Hopper <https://github.com/andyhopp>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+declare namespace connect {
+    /**
+     *
+     * A callback to receive agent details
+     *
+     * @param agent An Agent object containing information about the currently
+     * signed-in agent.
+     */
+    type AgentCallback = (agent: Agent) => void;
+
+    /**
+     * Register a callback to receive agent details
+     *
+     * @param callback A callback that will receive an {Agent} instance
+     *                 when agent information becomes available.
+     */
+    function agent(callback: AgentCallback): void;
+
+    /**
+     *
+     * A callback to receive contact details
+     *
+     * @param agent A Contact object containing information about the current contact.
+     */
+    type ContactCallback = (contact: Contact) => void;
+
+    /**
+     * Register a callback to receive contact details
+     *
+     * @param callback A callback that will receive a Contact instance
+     *                 when contact information is updated.
+     */
+    function contact(callback: ContactCallback): void;
+
+    /**
+     * Binds the given instance object as the context for
+     * the method provided.
+     *
+     * @param scope The instance object to be set as the scope
+     *              of the function.
+     * @param method The method to be encapsulated.
+     *
+     * All other arguments, if any, are bound to the method
+     * invocation inside the closure.
+     *
+     * @return A closure encapsulating the invocation of the
+     *    method provided in context of the given instance.
+     */
+    function hitch(scope: object, method: () => any): void;
+
+    interface Core {
+        initCCP(containerDiv: HTMLElement, options: InitCCPOptions): void;
+    }
+    let core: Core;
+
+    interface SoftPhoneOptions {
+        /*
+        * Whether to disable the ringtone.
+        */
+        disableRingtone?: boolean;
+        /*
+        * Whether to display the softphone in a frame.
+        */
+        allowFramedSoftphone?: boolean;
+        /*
+        * A URL for a custom ringtone.
+        */
+        ringtoneUrl?: string;
+     }
+
+    interface InitCCPOptions {
+        /*
+        * The URL for the Connect CCP.
+        */
+        ccpUrl: string;
+        /*
+        * Whether to display the login view.
+        */
+        loginPopup?: boolean;
+        /*
+        * Options specifying softphone configuration.
+        */
+        softphone?: SoftPhoneOptions;
+     }
+
+     enum AgentStateType {
```

Comment:
Is this really enum? Do the values like `AgentStateType` or `AgentAvailStates` etc exist at runtime. If not you want to declare this as union of string literal instead as :
```ts
type AgentStateType  = "init" | "routable" | "not_routable" | "offline";
```

Reply:
Hello! Yep, it is an enum; the expected usage per the ACS API docs is AgentStateType.INIT (ACS creates an enum object using the connect.makeEnum utility function).